server/bleep/src/agent.rs

```diff
@@ -181,6 +184,12 @@ impl Agent {
             Action::Proc { query, paths } => self.process_files(query, paths).await?,
         };
 
+        if self.exchanges.len() >= MAX_STEPS {
+            return Ok(Some(Action::Answer {
+                paths: self.paths().enumerate().map(|(i, _)| i).collect(),
```

Comment:
I haven't tested the answer quality, but it might be better if we get bloop itself to choose which paths to answer w.r.t.

E.g. If we exceed `MAX_STEPS` we force the agent to call the `none` function: https://platform.openai.com/docs/api-reference/chat/create#function_call